src/Automatonymous/AutomatonymousStateMachine.cs

```diff
@@ -431,17 +448,144 @@ void DeclareDataEvent<T>(PropertyInfo property)
             }
         }
 
+        protected internal virtual void CompositeEvent(string name,

```

Comment:
These will need to just be internal, I don't want them showing up as methods in state machine being declared in the constructor.